### PR TITLE
HFS+: Fix warnings and add function documentation

### PR DESCRIPTION
**Pull request by uckelman-sf**
This PR:

* enables more warnings for autotools builds (`-Wall -Wextra`)
* rectifies the confusion between `CPPFLAGS` and `CXXFLAGS` in various `Makefile.am`s
* fixes various compilation warnings
* adds documentation to the HFS+ decompression functions
* fixes a return value bug in `hfs_attr_walk_compressed_rsrc`

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -4,7 +4,7 @@ dnl Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
 
-AC_INIT(sleuthkit, 4.4.2)
```
Comment (sf-jonstewart) [inline configure.ac]: @uckelman-sf Does it make sense to patch this on the `develop` branch? TSK is already at 4.6 now.

Comment (uckelman-sf) [reply configure.ac]: Fixed now.

**Comment by uckelman-sf** [comment]
The two OS X builds are failing in Travis because the machines appear not to have `ant` installed.

**Comment by bcarrier** [comment]
@uckelman-sf , was there a specific warning that you were looking for when you added pedantic?   The output is much more verbose now (it's taking up most of the Travis CI output size) and I'm not sure how many are that relevant.

**Comment by bcarrier** [comment]
I changed develop to remove the -Wpedantic, but kept -Wextra (and suppressed the warnings about unused arguments).   The pedantic stuff is complaining about stuff that has never been an issue before on any platform that I'm aware of.  Was there a specific motivation?

**Pull request merged**